src/Infrastructure/Repository/TaskYamlRepository.php

```diff
@@ -54,38 +44,58 @@ class TaskYamlRepository implements TaskYamlRepositoryInterface
     protected Yaml $yamlParser;
 
     /**
-     * @var array<string, \SprykerSdk\SdkContracts\Entity\TaskInterface>
+     * @var \SprykerSdk\Sdk\Infrastructure\Builder\Yaml\TaskBuilderInterface
+     */
+    protected TaskBuilderInterface $taskBuilder;
+
+    /**
+     * @var \SprykerSdk\Sdk\Infrastructure\Builder\Yaml\TaskSetBuilderInterface
      */
-    protected array $existingTasks = [];
+    protected TaskSetBuilderInterface $taskSetBuilder;
+
+    /**
+     * @var \SprykerSdk\Sdk\Core\Application\Dependency\TaskPoolInterface
+     */
+    protected TaskPoolInterface $taskPool;
+
+    /**
+     * @var \SprykerSdk\Sdk\Core\Application\Dependency\TaskYamlFactoryInterface
+     */
+    protected TaskYamlFactoryInterface $taskYamlFactory;
 
     /**
      * @param \SprykerSdk\Sdk\Core\Application\Dependency\Repository\SettingRepositoryInterface $settingRepository
      * @param \Symfony\Component\Finder\Finder $fileFinder
      * @param \Symfony\Component\Yaml\Yaml $yamlParser
-     * @param iterable<\SprykerSdk\SdkContracts\Entity\TaskInterface> $existingTasks
+     * @param \SprykerSdk\Sdk\Infrastructure\Builder\Yaml\TaskBuilderInterface $taskBuilder
+     * @param \SprykerSdk\Sdk\Infrastructure\Builder\Yaml\TaskSetBuilderInterface $taskSetBuilder
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\TaskPoolInterface $taskPool
+     * @param \SprykerSdk\Sdk\Core\Application\Dependency\TaskYamlFactoryInterface $taskYamlFactory
      */
     public function __construct(
         SettingRepositoryInterface $settingRepository,
         Finder $fileFinder,
         Yaml $yamlParser,
-        iterable $existingTasks = []
+        TaskBuilderInterface $taskBuilder,
+        TaskSetBuilderInterface $taskSetBuilder,
+        TaskPoolInterface $taskPool,
+        TaskYamlFactoryInterface $taskYamlFactory
     ) {
         $this->yamlParser = $yamlParser;
         $this->fileFinder = $fileFinder;
         $this->settingRepository = $settingRepository;
-        foreach ($existingTasks as $existingTask) {
-            $this->existingTasks[$existingTask->getId()] = $existingTask;
-        }
+        $this->taskBuilder = $taskBuilder;
+        $this->taskSetBuilder = $taskSetBuilder;
+        $this->taskPool = $taskPool;
+        $this->taskYamlFactory = $taskYamlFactory;
     }
 
     /**
-     * @param array $tags
-     *
      * @throws \SprykerSdk\Sdk\Core\Application\Exception\MissingSettingException
      *
      * @return array
      */
-    public function findAll(array $tags = []): array
+    public function findAll(): array
```

Comment:
Looks much better, but still has something to improve.
It is not a repository. Here we have a couple of parts(which break the SRP principle)  - file parsing, task building, task set building, and updating the task registry.
Too many injected services.
Suggestion
1. rename (and move to proper directory) to the `TaskFileLoader`. 
2. extract task reading logic along with finder to the `TaskFileReader` which returns [$taskArrayCollection, $taskSetArrayCollection].
3. extract `foreach ($taskListData as $taskData)` to the separate method `collectTasks()`.
4. extract `foreach ($taskSetsData as $taskData)` to the separate method `collectTaskSets`.